### ISSUE TITLE
IAR compatibility - Linux sys files

### ISSUE THREAD

**Issue by back-40** [aws/aws-iot-device-sdk-embedded-C#213]
I'm using the embedded C sdk with IAR and having issues getting 'sys' dependency files to link properly. I installed GCC for ARM and am using its sys folder/files, but getting many errors related to INT, __INTPTR_TYPE__, etc defines missing.

Fatal Error[Pe035]: #error directive: "Unable to determine type definition of int32_t" C:\Program Files (x86)\GNU Tools Arm Embedded\7 2018-q2-update\arm-none-eabi\include\sys\_intsup.h 75 

Has anyone successfully used this with IAR/ARM? Any suggestions?
Thanks.

**Comment by gordonwang0** [comment]
Hi @back-40,

Are you sure that GNU Arm Embedded Toolchain headers are compatible with IAR?  My guess would be that they are not, as they are supposed to be used with GCC.

I'm not sure that the SDK as provided will build with IAR.  From my understanding, IAR is an embedded compiler for platforms that may not have an OS, while the samples provided with this SDK target Linux.  You will have to port the SDK to work with whatever platform you are using if it's not Linux.

**Comment by gordonwang0** [comment]
Hi @back-40,

Please let me know if you need any more help with this.  Closing issue.

**Comment by back-40** [comment]
Thanks @gordonwang0 , see about getting it ported.
